src/app/cluster/cluster-details/constraints/component.ts

```diff
@@ -97,11 +98,11 @@ export class ConstraintsComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   isLoadingData(data: Constraint[]): boolean {
-    return _.isEmpty(data) && !this.isClusterRunning;
+    return _.isEmpty(data) && !this.isClusterRunning && !this.hasMachineDeployments;
   }
 
   hasNoData(data: Constraint[]): boolean {
-    return _.isEmpty(data) && this.isClusterRunning;
+    return _.isEmpty(data) && this.isClusterRunning && this.hasMachineDeployments;
```

Comment:
What will happen if there is no data, cluster is running but has no machine deployments? I think due to the new checks you have added not all scenarios are covered. `hasNoData` will return false and the zero state message will not be displayed.

Reply:
Sorry, I've missed your comment.

Yes you are right. I thought about all cases again, and I think the machine deployments check isn't necessary at all for this views. data + cluster running is the important part and it doesn't change anything if there are machine deployments or not 👍